docs/concepts.md

```diff
@@ -1,27 +1,213 @@
-## Concepts
+# Concept Reference
 
-The ksonnet framework is centered around a couple of key concepts.
+The ksonnet framework is centered around a couple of key concepts and terms. This reference gives a brief explanation of each. To get a more tangible sense of how these concepts tie together, see the Tutorial (WIP, link pending).
+
+* Ksonnet
+  * [Application](#application)
+  * [Environment](#environment)
+  * [Component](#component)
+  * [Prototype](#prototype)
+  * [Parameter](#parameter)
+  * [Part](#part)
+  * [Package](#package)
+  * [Registry](#registry)
+* Related
+  * [Manifest](#manifest)
+  * [Jsonnet](#jsonnet)
+
+
+The following diagram shows how the ksonnet framework works holistically:
+
+![ksonnet overview diagram](/docs/img/ksonnet_overview.svg)
+
+---
 
 ### Application
-TODO
+
+A ksonnet application represents a well-structured directory of Kubernetes [manifests](#manifests). This directory is autogenerated by [`ksonnet init`](docs/cli-reference/ksonnet_init.md). These manifests typically tie together in some way—collectively, they might define something like the following:
```

Comment:
Would be helpful here I think to give some examples of what they define together, or call out what that diagram actually is.

`... like the following web service:`